cli/src/pcluster/validators/directory_service_validators.py

```diff
@@ -82,37 +82,20 @@ def _validate(self, password_secret_arn: str, region: str):
             for retro-compatibility.
         """
         try:
-            # We only require the secret to exist; we do not validate its content.
-            arn_components = password_secret_arn.split(":")
-            service = arn_components[2]
-            resource = arn_components[5]
+            service, resource = _get_service_and_resource(password_secret_arn)
             if service == "ssm":
-                resource = arn_components[5].split("/")[0]
-
+                resource = password_secret_arn.split(":")[5].split("/")[0]
```

Comment:
Because the ssm ARN and the secretsmanager ARN have different format.

The secrets manager ARN follows following format:
`arn:aws:secretsmanager:region:account-id:secret:secret-name`

The ssm service ARN format is following:
`arn:aws:ssm:region:account-id:resource-type/resource-id`
`resource-type` is the type of SSM resource, eg: `parameter` or `document`
So The ARN for an SSM parameter might look like this:
arn:aws:ssm:us-west-2:123456789012:parameter/parameter-name

Reply:
Thank you for the explanation. Two comments/questions here:

- this maybe means that the function `get_arn_service_and_resource()` introduced above is already not general or useful enough? We will see that in the future.
- on line 92 I would do:
  ```python3
  resource = resource.split("/")[0]
  ```
  what do you think?

Reply:
Done!